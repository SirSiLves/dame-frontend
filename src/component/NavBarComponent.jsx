import React, {Component} from 'react';
import goBack from "../goback.png";
import '../stylesheet/toggleSwitch.css'


class NavBarComponent extends Component {

    constructor(props) {
        super(props);
        this.botEnabledCheckbox = React.createRef();
    }

    render() {
        return (
            <nav className="navbar navbar-light bg-light">
                <div className="container justify-content-center">
                    <div className="pr-0">
                        <h1>Let's play a game</h1>
                    </div>
                    <div className="d-flex mt-2 align-items-center">
                        <div className="p-2">
                            <button className="btn btn-dark" onClick={this.props.onStartDefault}>
                                8 x 8
                            </button>
                        </div>
                        <div className="p-0">
                            <h1>/</h1>
                        </div>
                        <div className="p-2">
                            <button className="btn btn-dark" onClick={this.props.onStartExtend}>
                                10 x 10
                            </button>
                        </div>
                        <div className="d-flex flex-wrap">
                            <div className="p-0 p-md-2 ml-1 ml-md-0 mt-1">
                                <label className="switch">
                                    <input className="switch-input" type="checkbox" ref={this.botEnabledCheckbox}
                                           onClick={() => this.props.botEnabledMode(this.botEnabledCheckbox.current.checked)}/>
                                    <span className="switch-label" data-on="Bot On" data-off="Bot Off" />
                                    <span className="switch-handle" />
                                </label>
                            </div>
                            <div className="p-0 p-md-2 ml-2 ml-md-0 mb-1 mt-1">
                                <img src={goBack} alt={"Revert move"} height={40} width={40} onClick={this.props.onRevertMove}/>
                            </div>
                        </div>
                    </div>
                </div>
            </nav>
        );
    }
}

export default NavBarComponent;
