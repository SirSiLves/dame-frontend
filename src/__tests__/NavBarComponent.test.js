import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NavBarComponent from "../component/NavBarComponent";


it('renders NavBarComponent without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NavBarComponent />, div);
});

describe('NavBarComponent Component', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<NavBarComponent debug />);

        expect(component).toMatchSnapshot();
    });
});
