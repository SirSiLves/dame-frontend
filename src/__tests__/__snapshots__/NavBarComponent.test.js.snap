// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarComponent Component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
